test/integration/controller/core/clusterqueue_controller_test.go

```diff
@@ -362,6 +362,63 @@ var _ = ginkgo.Describe("ClusterQueue controller", ginkgo.Ordered, ginkgo.Contin
 			util.ExpectReservingActiveWorkloadsMetric(clusterQueue, 0)
 		})
 
+		ginkgo.It("Should update status and report metrics when a pending workload is deleted", func() {
+			workload := testing.MakeWorkload("one", ns.Name).Queue(localQueue.Name).
+				Request(corev1.ResourceCPU, "5").Obj()
+
+			ginkgo.By("Creating a workload", func() {
+				gomega.Expect(k8sClient.Create(ctx, workload)).To(gomega.Succeed())
+			})
+
+			// Pending workloads count is incremented as the workload is inadmissible
```

Comment:
non-blocking nit: I would prefer to turn the comment into ginkgo step. Maybe ginkgo.By("await until the pending workload count is incremented as the workload is inadmissible, because the ResourceFlavors don't exist"). This way the comment is also logged, which can make debugging easier.

Reply:
Agreed. Actually I saw an existing comment and somehow decided to choose that style, but I would have chosen the `ginkgo.By` stanza otherwise. Let me know if you feel like it's worth I open a follow-up PR.

Reply:
I didn't notice the PR was approved actually, and though you would have time to apply the remark. I'm ok either way, as you feel.